### PR TITLE
fix build base bug

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Fixes an issue with build not respecting `base` config

## Testing

- Updated tests.

## Docs

- N/A

### CODE REVIEW

**Comment by FredKSchott** [comment]
Self-merging as steward since core team is out today, and this is small.

File: packages/astro/src/core/build/generate.ts [hunk 1]
```diff
@@ -191,7 +191,7 @@ async function generatePath(
 	}
 
 	const ssr = isBuildingToSSR(opts.astroConfig);
-	const url = new URL(origin + pathname);
+	const url = new URL(opts.astroConfig.base + pathname.substring(1), origin);
```
Comment (tony-sull) [inline packages/astro/src/core/build/generate.ts:194]: I think the answer here is yes, but is it safe to always assume `pathname` will start with a `/` here?

If not, may be a good spot for a regex that only strips the first character if it's a `/`

**Review by tony-sull — APPROVED** [review]
Looks good to me!  Added a question just as a sanity check but this fix + test should have it covered :+1:

File: packages/astro/src/core/build/generate.ts (hunk 1 above)
Comment (FredKSchott) [reply packages/astro/src/core/build/generate.ts:194]: yea, agreed. better safe than sorry

**Comment by FredKSchott** [comment]
LGTM!

**Pull request merged**